vitrina/datasets/managers.py

```diff
@@ -1,6 +1,7 @@
 from django.db import models
+from django.db.models import Q
 
 
 class PublicDatasetManager(models.Manager):
     def get_queryset(self):
-        return super().get_queryset()
+        return super().get_queryset().filter(is_public=True).filter(~Q(deleted=True))
```

Comment:
This is not, how it works currently. Here is a code sample, how datasets are filtered now:

```java
        if (d.getDeleted() != null || d.getDeletedOn() != null) {
            return;
        }

        if (d.getPublic() == null || !d.getPublic()) {
            return;
        }

        // todo check this one
        if (d.getOrganizationId() != null) {
            Optional<Organization> byId = organizationRepository.findById(d.getOrganizationId());
            if (byId.isPresent()) {
                if (byId.get().getPublic() != null) {
                    if (!byId.get().getPublic()) {
                        return;
                    } else {
                        indexOrganization(byId.get());
                        organizationCore.commit();
                    }
                }
            }
        }
```

Also you must always handle NULL values explicitly, because if you don't they will be ignored and will show up in the results, so correct query would be `.filter(deleted__isnull=False)`.

And please, add tests for this, because this is second incorrect attempt.